ota_pal.c

```diff
@@ -75,12 +93,183 @@ static psa_image_id_t xOTAImageID = TFM_FWU_INVALID_IMAGE_ID;
+
+        /********* R Component. *********/
+
+        /* R components are represented by mbedTLS as 33 bytes when the first bit is zero to avoid any sign confusion. */
+        if( ucSigComponentLength == 33UL )
+        {
+            /* Chop off the leading zero.  The first 4 bytes were SEQUENCE, LENGTH, R_INTEGER, R_LENGTH, 0x00 padding.  */
+            ( void ) memcpy( pucRawSignature, &pucEncodedSignature[ 5 ], 32 );
+            /* SEQUENCE, LENGTH, R_INTEGER, R_LENGTH, leading zero, R, S's integer tag */
+            pxNextLength = &pucEncodedSignature[ 5U + 32U + 1U ];
+        }
+        else if( ucSigComponentLength <= 32UL )
+        {
+            /* The R component is 32 bytes or less.  Copy so that it is properly represented as a 32 byte value,
+             * leaving leading 0 pads at beginning if necessary. */
+            ( void ) memcpy( &pucRawSignature[ 32UL - ucSigComponentLength ],  /* If the R component is less than 32 bytes, leave the leading zeros. */
+                             &pucEncodedSignature[ 4 ],                        /* SEQUENCE, LENGTH, R_INTEGER, R_LENGTH, (R component begins as the 5th byte) */
+                             ucSigComponentLength );
+            pxNextLength = &pucEncodedSignature[ 4U + ucSigComponentLength + 1U ]; /* Move the pointer to get rid of
+                                                                                    * SEQUENCE, LENGTH, R_INTEGER, R_LENGTH, R Component, S_INTEGER tag. */
+        }
+        else
+        {
+            xReturn = false;
+        }
+
+        /********** S Component. ***********/
+
+        if( xReturn == true )
+        {
+            /* Now pxNextLength is pointing to the length of the S component. */
+            ucSigComponentLength = pxNextLength[ 0 ];
+
+            if( ucSigComponentLength == 33UL )
+            {
+                ( void ) memcpy( &pucRawSignature[ 32 ],
+                                 &pxNextLength[ 2 ], /* S_LENGTH (of S component), 0x00 padding, S component is 3rd byte - we want to skip the leading zero. */
+                                 32 );
+            }
+            else if( ucSigComponentLength <= 32UL )
+            {
+                /* The S component is 32 bytes or less.  Copy so that it is properly represented as a 32 byte value,
+                 * leaving leading 0 pads at beginning if necessary. */
+                ( void ) memcpy( &pucRawSignature[ 64UL - ucSigComponentLength ],
+                                 &pxNextLength[ 1 ],
+                                 ucSigComponentLength );
+            }
+            else
+            {
+                xReturn = false;
+            }
+        }
+    }
+
+    return xReturn;
+}
+
+static OtaPalStatus_t prvCheckECDSASignature( OtaFileContext_t * const pFileContext )
+{
+#if defined( OTA_PAL_CODE_SIGNING_ALGO ) && ( OTA_PAL_CODE_SIGNING_ALGO == OTA_PAL_CODE_SIGNING_RSA )
```

Comment:
Because ucECDSARAWSignature is used in prvCheckECDSASignature, and it's declared only when OTA_PAL_CODE_SIGNING_ALGO is set to OTA_PAL_CODE_SIGNING_ECDSA.